packages/superset-ui-plugin-chart-table/src/Table.tsx

```diff
@@ -0,0 +1,206 @@
+import DataTable from '@airbnb/lunar/lib/components/DataTable';
+import Text from '@airbnb/lunar/lib/components/Text';
+import Input from '@airbnb/lunar/lib/components/Input';
+import withStyles, { css, WithStylesProps } from '@airbnb/lunar/lib/composers/withStyles';
+import { Renderers, ParentRow, ColumnMetadata } from '@airbnb/lunar/lib/components/DataTable/types';
+import { getRenderer, ColumnType, heightType, Cell } from './renderer';
+
+type Props = {
+  data: ParentRow[];
+  height: number;
+  alignPositiveNegative?: boolean;
+  colorPositiveNegative?: boolean;
+  columns: ColumnType[];
+  filters?: {
+    [key: string]: any[];
+  };
+  includeSearch?: boolean;
+  onAddFilter?: (key: string, value: number[]) => void;
+  onRemoveFilter?: (key: string, value: number[]) => void;
+  tableFilter: boolean;
+};
+
+function NOOP(key: string, value: number[]) {}
+
+const defaultProps = {
+  alignPositiveNegative: false,
+  colorPositiveNegative: false,
+  filters: {},
+  includeSearch: false,
+  onAddFilter: NOOP,
+  onRemoveFilter: NOOP,
+};
+
+export type TableProps = Props & Readonly<typeof defaultProps>;
+
+type InternalTableProps = TableProps & WithStylesProps;
+
+type TableState = {
+  selectedCells: Set<string>;
+  searchKeyword: string;
+  filteredRows: ParentRow[];
+};
+
+function getCellHash(cell: Cell) {
+  return `${cell.key}#${cell.value}`;
+}
+
+class TableVis extends React.PureComponent<InternalTableProps, TableState> {
+  static defaultProps = defaultProps;
+
+  constructor(props: InternalTableProps) {
+    super(props);
+    this.state = {
+      selectedCells: new Set(),
+      searchKeyword: '',
+      filteredRows: [],
+    };
+  }
+
+  handleCellSelected = (cell: Cell) => () => {
+    const { selectedCells } = this.state;
+    const { tableFilter, onRemoveFilter, onAddFilter } = this.props;
+
+    if (!tableFilter) {
+      return;
+    }
+    const newSelectedCells = new Set(Array.from(selectedCells));
+    const cellHash = getCellHash(cell);
+    if (newSelectedCells.has(cellHash)) {
+      newSelectedCells.delete(cellHash);
+      onRemoveFilter(cell.key, [cell.value as number]);
+    } else {
+      newSelectedCells.add(cellHash);
+      onAddFilter(cell.key, [cell.value as number]);
+    }
+    this.setState({
+      selectedCells: newSelectedCells,
+    });
+  };
+
+  isSelected = (cell: Cell) => {
+    const { selectedCells } = this.state;
+    return selectedCells.has(getCellHash(cell));
+  };
+
+  handleSearch = (value: string) => {
+    const { searchKeyword } = this.state;
+    const { data } = this.props;
+    if (searchKeyword !== value) {
+      const filteredRows = data.filter(row => {
+        const content = Object.values(row.data)
+          .join('|')
+          .toLowerCase();
+        return content.indexOf(value) >= 0;
+      });
+      this.setState({
+        searchKeyword: value,
+        filteredRows,
+      });
+    }
+  };
+
+  static getDerivedStateFromProps: React.GetDerivedStateFromProps<
```

Comment:
@conglei I just meant that this is called every time that props change.

So if `filters` have not changed between updates, you wouldn't need to update state right? (in that case you can return `null` from this func). If I'm misunderstanding you can ignore this :)

Reply:
Ah, gotcha, good catch! will revise the logic :)